decidim-bulletin_board-js/src/client/message-parser.js

```diff
@@ -0,0 +1,99 @@
+import { JWS, JWK } from "node-jose";
+import {
+  MessageIdentifier,
+  AUTHORITY_TYPE,
+  BULLETIN_BOARD_TYPE,
+  TRUSTEE_TYPE,
+} from "./message-identifier";
+import { samePublicKeys } from "../utils";
+
+/**
+ * Verify and parses JWT tokens.
+ */
+export class MessageParser {
+  /**
+   * Initializes the class with the given params.
+   *
+   * @constructor
+   * @param {Object} params - An object that contains the initialization params.
+   * - {String} authorityPublicKeyJSON - The public key of the authority
+   */
+  constructor({ authorityPublicKeyJSON }) {
+    this.authorityPublicKeyJSON = authorityPublicKeyJSON;
+    JWK.asKey(authorityPublicKeyJSON, "json").then((result) => {
+      this.authorityPublicKey = result;
+    });
```

Comment:
This is a bit "dangerous" since you are assigning that value when the promise is resolved and you don't know when that will be. I propose you an alternative 👍 

```js
this.authorityPublicKey = JWK.asKey(authorityPublicKeyJSON, "json");
```

Then, when you want to use that in the code you do something like this:

```js
async myFunction() {
  // ...
  const authorityPublicKey = await this.authorityPublicKey;
  // ...
}
```

This will make sure that you have an actual value to work with 😄

Reply:
Thanks! I had this pending to be resolved and your solution is exactly what I needed 😍